### DISCUSSION TITLE
KQL query for seeing all operations in Durable Functions

### DISCUSSION THREAD

**Discussion by Nibushi** [Azure/azure-functions-durable-extension#1940] [Q&A] [answered]
Hi

I am trying to troubleshoot an intermittent issue with my durable function. The function is being called from a Power Automate flow. The majority of the time the function succeeds as expected, however, other times it looks like the Power Automate flow has called the function but it never gets a response and just keeps running. 

I am trying to match up the approx. time for the Flow call to the Orchestrator function in the Azure logs. I am trying to figure out if there is a way to build a list of all the operations carried out by an execution of a durable function based on the parent/child relationships of the operation IDs in the Azure logs?

This might be completely the wrong way to do this, so if there is a better way to see all the operations associated with a call to the Durable functions' Orchestrator function I'd be really grateful to hear it. 

Any help would be appreciated.

Cheers

**Comment by cgillum** [comment, marked as answer]
The simplest way to do this is to use query the Application Insights logs emitted by the orchestration, as shown here: https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-diagnostics?tabs=csharp#application-insights. The documentation shows a few example KQL queries you can use for this.

However, this might not be exactly what you're asking. For example, if you're trying to do end-to-end correlation, we have a feature for it, but it's not 100% complete yet. You can find more information, including a sample, here: https://github.com/Azure/azure-functions-durable-extension/tree/dev/samples/correlation-csharp#readme
